### ISSUE TITLE
Removing parse method/endpoint

### ISSUE THREAD

**Issue by joshmosh** [davidcelis/geocodio#11]
According to the geocod.io docs as of June 2015 the parse endpoint has been deprecated. Geocode is now returning parsed address fields so maybe we should remove that from the gem as well.

**Comment by davidcelis** [comment]
👍 Seems good to me. Do you want to submit a PR to remove them and their associated tests?

**Comment by joshmosh** [comment]
Of course, I'd love to help contribute to this. Since they haven't completely removed the endpoint I think we're safe to leave it for now. I do however think a good plan of attack would be to attach the parsed address fields to the `AddressSet` class like `query` is currently. Here's an example response from a reset test I did:

```json
{
  "input": {
    "address_components": {
      "number": "1100",
      "predirectional": "S",
      "street": "Lamar",
      "suffix": "Blvd",
      "formatted_street": "S Lamar Blvd",
      "city": "Austin",
      "state": "TX",
      "zip": "78704",
      "country": "US",
      "secondaryunit": "Apt",
      "secondarynumber": "1549"
    },
    "formatted_address": "1100 S Lamar Blvd, Austin, TX 78704"
  },
  "results": [
    {
      "address_components": {
        "number": "1100",
        "predirectional": "S",
        "street": "Lamar",
        "suffix": "Blvd",
        "formatted_street": "S Lamar Blvd",
        "city": "Austin",
        "county": "Travis County",
        "state": "TX",
        "zip": "78704",
        "country": "US"
      },
      "formatted_address": "1100 S Lamar Blvd, Austin, TX 78704",
      "location": {
        "lat": 30.255551,
        "lng": -97.761915
      },
      "accuracy": 1,
      "accuracy_type": "range_interpolation",
      "source": "TIGER\/Line\u00ae dataset from the US Census Bureau"
    },
    {
      "address_components": {
        "number": "1100",
        "predirectional": "S",
        "street": "Lamar",
        "suffix": "Blvd",
        "formatted_street": "S Lamar Blvd",
        "city": "Austin",
        "county": "Travis County",
        "state": "TX",
        "zip": "78704",
        "country": "US"
      },
      "formatted_address": "1100 S Lamar Blvd, Austin, TX 78704",
      "location": {
        "lat": 30.25542,
        "lng": -97.761645
      },
      "accuracy": 0.8,
      "accuracy_type": "range_interpolation",
      "source": "TIGER\/Line\u00ae dataset from the US Census Bureau"
    }
  ]
}
```

You'll notice the first part of the response has a key for `"input"` which shows the query parsed out nicely. My thinking is this could be access like so:

```ruby
location = geocodio.geocode(['1100 S Lamar Blvd Apt 1549 Austin TX 78704'])

location.input # => <Geocodio::Address: ...>
```

The only real difference between the address class now and the input coming in is the two extra fields for unit types and numbers.

- secondaryunit
- secondarynumber

How do you feel about adding those fields to the address class and reusing it as the `location.input` as shown above versus creating a new class like `Input` or `Parsed` under the Geocodio module that would be used for storing address input information?

I hope I'm making sense. Let me know if I need to clarify anything.

**Comment by davidcelis** [comment]
@joshmosh That sounds like a great idea. I think including `secondaryunit` and `secondarynumber` into `Address` makes sense either way. I'd only prefer that the attributes get defined as `secondary_unit` and `secondary_number` to be more semantic, but otherwise, big 👍!

Only other suggestion I'd make is that we can add a `warn` message about the parse endpoint being deprecated if we leave the actual `parse` method in place. The message can read something like "Warning: Geocodio's `parse` endpoint is deprecated and will be removed in a future version. Please use `geocode` instead and use the result's `input` attribute to retrieve a parsed address."

**Comment by joshmosh** [comment]
@davidcelis My thoughts exactly all around. Great idea for the warning message. I may need your help with that piece.

**Comment by davidcelis** [comment]
@joshmosh Should be enough to just call `warn("message")` (it's a Kernel method, so it's available everywhere) but happy to help wherever needed.

**Comment by joshmosh** [comment]
@davidcelis Thanks for the help!
